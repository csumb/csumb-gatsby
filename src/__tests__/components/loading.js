import React from 'react'
import renderer from 'react-test-renderer'
import Loading from '../../components/common/loading'

describe('Loading', () => {
  it('renders a loading indicator correctly', () => {
    const tree = renderer.create(<Loading>I am loading!</Loading>).toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('renders a loading indicator correctly without children', () => {
    const tree = renderer.create(<Loading />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
