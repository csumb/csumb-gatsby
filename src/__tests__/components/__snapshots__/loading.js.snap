// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading renders a loading indicator correctly 1`] = `
<div
  className="css-1ytd9z0"
>
  <span>
    I am loading!
  </span>
  <div
    dangerouslySetInnerHTML={
      Object {
        "__html": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 32 32\\">
  <path opacity=\\".25\\" d=\\"M16 0 A16 16 0 0 0 16 32 A16 16 0 0 0 16 0 M16 4 A12 12 0 0 1 16 28 A12 12 0 0 1 16 4\\"/>
  <path d=\\"M16 0 A16 16 0 0 1 32 16 L28 16 A12 12 0 0 0 16 4z\\">
    <animateTransform attributeName=\\"transform\\" type=\\"rotate\\" from=\\"0 16 16\\" to=\\"360 16 16\\" dur=\\"0.8s\\" repeatCount=\\"indefinite\\" />
  </path>
</svg>
",
      }
    }
    style={
      Object {
        "fill": "#112e51",
        "height": 30,
        "width": 30,
      }
    }
  />
</div>
`;

exports[`Loading renders a loading indicator correctly without children 1`] = `
<div
  className="css-1ytd9z0"
>
  <span>
    <div
      className="css-1p349jc"
    >
      Loading
    </div>
  </span>
  <div
    dangerouslySetInnerHTML={
      Object {
        "__html": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 32 32\\">
  <path opacity=\\".25\\" d=\\"M16 0 A16 16 0 0 0 16 32 A16 16 0 0 0 16 0 M16 4 A12 12 0 0 1 16 28 A12 12 0 0 1 16 4\\"/>
  <path d=\\"M16 0 A16 16 0 0 1 32 16 L28 16 A12 12 0 0 0 16 4z\\">
    <animateTransform attributeName=\\"transform\\" type=\\"rotate\\" from=\\"0 16 16\\" to=\\"360 16 16\\" dur=\\"0.8s\\" repeatCount=\\"indefinite\\" />
  </path>
</svg>
",
      }
    }
    style={
      Object {
        "fill": "#112e51",
        "height": 30,
        "width": 30,
      }
    }
  />
</div>
`;
